drivers/ethernet/eth_mcux.c

```diff
@@ -898,8 +902,11 @@ static inline void ts_register_tx_event(struct eth_context *context,
 			if (frameinfo->isTsAvail) {
 				k_mutex_lock(&context->ptp_mutex, K_FOREVER);
 
+				/* The offset added corresponds to the time it takes for the packet 
+				* to transit between the mac and the phy 
+				*/
 				pkt->timestamp.nanosecond =
-					frameinfo->timeStamp.nanosecond;
+					frameinfo->timeStamp.nanosecond+840;
```

Comment:
And here maybe:
if (pkt->timestamp.nanosecond >= 1000000000) {
pkt->timestamp.nanosecond -= 1000000000;
pkt->timestamp.second++;
}

Reply:
there is also some logic a little higher up checking for boundaries so wondering if this needs to be accounted for?